interop_binaries/build.rs

```diff
@@ -0,0 +1,62 @@
+fn main() {
+    // We only build the container image if the `testcontainer` feature is enabled, in order to
+    // avoid infinite recursion in our build process (since building the container image builds this
+    // package, among other things.)
+
+    #[cfg(feature = "testcontainer")]
+    {
+        use std::{env, process::Command};
+
+        // Build & save off a container image for the interop_aggregator.
+        // Note: `docker build` has an `--output` flag which writes the output to somewhere, which
+        // may be a tarfile. But `docker build --output` only exports the image filesystem, and not
+        // any other image metadata (such as exposed ports, the entrypoint, etc), so we can't easily
+        // use it.
+        let build_output = Command::new("docker")
+            .args([
+                "build",
+                "--file=Dockerfile.interop_aggregator",
```

Comment:
I pulled in the Dockerfile itself & the dependencies it copies into these directives. (I had naively thought that not specifying directives would cause the script to be rerun if anything in the workspace changed, but re-reading the docs, I think the default is to re-run the script if anything in the package changes -- which isn't what we want.)

[While doing this, I noticed we were copying `monolithic_integration_test` into the build layer for the interop binaries; I removed this as it doesn't appear to be necessary, but calling it out in case this is undesirable.]

Reply:
That was generally required because cargo would read `Cargo.toml`, see the list of workspace members, and read `Cargo.toml` from each subdirectory, plus check for `src/lib.rs` or `src/main.rs`. I'm not sure if we can get away without copying it in or rewriting files, but if so, sounds good.